ksqldb-api-client/src/main/java/io/confluent/ksql/api/client/impl/DdlDmlResponseHandlers.java

```diff
@@ -33,14 +35,24 @@ private DdlDmlResponseHandlers() {
 
   static void handleExecuteStatementResponse(
       final JsonObject ksqlEntity,
-      final CompletableFuture<Void> cf
+      final CompletableFuture<ExecuteStatementResult> cf
   ) {
     if (!isCommandStatusEntity(ksqlEntity)) {
       handleUnexpectedEntity(ksqlEntity, cf);
       return;
     }
 
-    cf.complete(null);
+    try {
+      final JsonObject commandStatus = ksqlEntity.getJsonObject("commandStatus");
+      final Optional<String> queryId = commandStatus.getString("queryId") != null
```

Comment:
I think you can replace this with Optional.ofNullable(commandStatus.getString("queryId"));

Reply:
Ah, thanks for the tip!